### PR TITLE
feature (TerraForm): creating a wrapper component for the select box

### PR DESCRIPTION
**Pull request by zhajjhassan**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser-Support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser-Support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# 11.2.0 (XX.XX.2021)
+
+### Features
+
+-   **terra-form**
+    -   **text-select** replaced deprecated `terra-select-box` with a `mat-select` variant.
```
Comment (pweyrich) [inline CHANGELOG.md:6]:
```suggestion
    -   **select** replaced deprecated `terra-select-box` with a `mat-select` variant.
```

File: src/lib/components/forms/form/form-components/select/select-box.component.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<mat-form-field
+    class="w-100 form-input"
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.html:2]:
```suggestion
    class="w-100"
```
Die Klasse gibt es nicht.

File: src/lib/components/forms/form/form-components/select/select-box.component.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    [tcTooltip]="tooltipText"
+    [placement]="tooltipPlacement"
+    [onlyEllipsisTooltip]="true"
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.html:5]: Wieso?

File: src/lib/components/forms/form/form-components/select/select-box.component.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+>
+    <mat-label attr.for="{{ name }}">
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.html:7]:
```suggestion
    <mat-label>
```
Das regelt material selbst

File: src/lib/components/forms/form/form-components/select/select-box.component.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Input()
+    public tooltipText: string;
+
+    /** @description Set the tooltip placement (bottom, top, left, right). Default bottom. */
+    @Input()
+    public tooltipPlacement: TerraPlacementEnum = TerraPlacementEnum.BOTTOM;
+
+    @Input()
+    public selectBoxValues: Array<TerraSelectBoxValueInterface>; // TODO SelectInterface
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.ts:38]:
```suggestion
    public listBoxValues: Array<TerraSelectBoxValueInterface>; // TODO SelectInterface
```

File: src/lib/components/forms/form/form-components/select/select-box.component.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    <mat-select
+        [(ngModel)]="_value"
+        (ngModelChange)="_onChangeCallback($event)"
+        (blur)="_onTouchedCallback()"
+        [required]="isRequired"
+        [disabled]="isDisabled"
+    >
+        <mat-option *ngFor="let item of selectBoxValues" [value]="item?.value">
+            {{ item?.caption }}
+        </mat-option>
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.html:19]: Sortierung nach `position`, ein optionales icon und die Möglichkeit zur Einfärbung des Textes fehlt hier noch. Siehe `TerraSelectBoxInterface`

File: src/lib/components/forms/form/form-components/select/select-box.component.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    templateUrl: './select-box.component.html',
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: SelectBoxComponent,
+            multi: true
+        }
+    ]
+})
+export class SelectBoxComponent {
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.ts:19]:
```suggestion
    ]
})
export class SelectBoxComponent implements SelectInterface {
```

File: src/lib/components/forms/form/form-components/select/select-box.component.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    /** @description The name of the select box also used to set the label. */
+    @Input()
+    public name: string = ' ';
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.ts:22]:
```suggestion
    public name: string = '';
```

File: src/lib/components/forms/form/form-components/select/select-box.component.ts [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Input()
+    public isRequired: boolean;
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.ts:25]:
```suggestion
    public isRequired: boolean = false;
```

File: src/lib/components/forms/form/form-components/select/select-box.component.ts [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Input()
+    public isDisabled: boolean;
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.ts:28]:
```suggestion
    public isDisabled: boolean = false;
```

File: src/lib/components/forms/form/form-components/select/select-box.component.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    public tooltipText: string;
+
+    /** @description Set the tooltip placement (bottom, top, left, right). Default bottom. */
+    @Input()
+    public tooltipPlacement: TerraPlacementEnum = TerraPlacementEnum.BOTTOM;
+
+    @Input()
+    public selectBoxValues: Array<TerraSelectBoxValueInterface>; // TODO SelectInterface
+
+    public _value: number | string;
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/select/select-box.component.ts:40]:
```suggestion
    public _value: any;
```

File: src/lib/components/forms/form/form-components/select/select.component.ts [hunk 12]
```diff
@@ -15,30 +16,44 @@ import { noop } from 'rxjs';
...
-    public isRequired: boolean;
+    public isDisabled: boolean = false;
 
     @Input()
-    public name: string;
+    public isRequired: boolean = false;
 
     @Input()
-    public tooltipPlacement: string;
+    public tooltipText: string;
```
Comment (nappynapster) [inline src/lib/components/forms/form/form-components/select/select.component.ts:31]: jsdoc bitte noch für die restlichen inputs

**Pull request merged**